cni/pkg/repair/netns.go

```diff
@@ -0,0 +1,150 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repair
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"strconv"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	"github.com/prometheus/procfs"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	corev1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/pkg/log"
+)
+
+func getPidNamespace(pid int) string {
+	return "/host/proc/" + strconv.Itoa(pid) + "/ns/net"
+}
+
+func runInHost[T any](f func() (T, error)) (T, error) {
```

Comment:
nit: here we donot need return value, and no generic. For those callers that need return value, it can be set in the closure.